agent-inject/agent/config.go

```diff
@@ -126,6 +120,16 @@ func (a *Agent) newConfig(init bool) ([]byte, error) {
 		Templates: a.newTemplateConfigs(),
 	}
 
+	if a.Vault.AuthMethod == "" || a.Vault.AuthMethod == DefaultVaultAuthMethod {
```

Comment:
Wouldn't it better to abstract this logic rather than having it all over the place? If were to do it I would create an interface with methods `GetAutoAuthConfig` and `GetAutoAuthPatches` and create structs like `KubernetesAutoAuth` and `CertAutoAuth` which implements this interface. I can see this function growing as now there are 2 open PRs for adding different autoauthmethods soon there will be more autoauth methods and we will end up with a very big function. Apart from cognitive overhead it would also make it hard to test this method in future. What do you think?

Reply:
I totally agree, however that was never the scope of this. At the time of implementation we had the need to support authentication using certificates. We don't forsee any need for anything else in the future and there seemed to be no efforts in the upstream implementation to provide an abstract version.
If you feel this is not up to the projects standards it is perfectly fine to dismiss this PR, there will be no hard feelings. We don't use this solution anymore and will not put more time into it.